src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -120,24 +123,28 @@ static MsQuicApi()
         private static bool IsWindowsVersionSupported() => OperatingSystem.IsWindowsVersionAtLeast(MinWindowsVersion.Major,
             MinWindowsVersion.Minor, MinWindowsVersion.Build, MinWindowsVersion.Revision);
 
-        private static bool IsTls13Disabled()
+        private static bool IsTls13Disabled(bool isServer)
         {
 #if TARGET_WINDOWS
-            string[] SChannelTLS13RegKeys = {
-                @"SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client",
-                @"SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Server"
-            };
+            string SChannelTls13RegistryKey = isServer
+                ? @"SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Server"
+                : @"SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client";
+
+            using var regKey = Registry.LocalMachine.OpenSubKey(SChannelTls13RegistryKey);
 
-            foreach (var key in SChannelTLS13RegKeys)
+            if (regKey is null)
             {
-                using var regKey = Registry.LocalMachine.OpenSubKey(key);
+                return false;
+            }
 
-                if (regKey is null) return false;
+            if (regKey.GetValue("Enabled") is int enabled && enabled == 0)
+            {
+                return true;
+            }
 
-                if (regKey.GetValue("Enabled") is int enabled && enabled == 0)
-                {
-                    return true;
-                }
+            if (regKey.GetValue("DisabledByDefault") is int disabled && disabled == 1)
```

Comment:
With this change, `DisabledByDefault=1` with `Enabled=1` is treated as disabled, is that correct? I would expect that `Enabled=1` would overrule `DisabledByDefault`.

Reply:
In fact this configuration is explicitly referred to in the documentation:

> In order to override a system default and set a supported (D)TLS or SSL protocol version to the Disabled by default state, create DWORD registry values named "Enabled" and "DisabledByDefault" with a non-zero value under the corresponding version-specific subkey. The following example shows TLS 1.0 server set to the Disabled by default state:
>
> ![image](https://user-images.githubusercontent.com/64654/174282828-ec9323ec-de34-436c-9db2-ed032ce38838.png)

https://docs.microsoft.com/en-us/windows-server/security/tls/tls-registry-settings

Reply:
I tried this combination locally when debugging the original issue and it still led to `SEC_E_ALGORITHM_MISMATCH`

Reply:
That may also depend on protocols we use e.g. `SslProtocols.None` vs `SslProtocols.Tls13`. One defers to OS default and one explicitly asks.